// cutie-ui/variables.js
const initialVariables = {
  background: '#ffffff',
  backgroundBlur: 'rgba(245, 245, 245, 0.9)',
  backgroundSecondary: 'rgba(245, 245, 245)',
  baseBorderRadius: '0.435rem',
  baseBoxShadow: '0px 0px 10px 0px rgba(0, 0, 0, 0.15)',
  baseFontFamily: '"system-ui", sans-serif',
  black: '#000',
  darkGlass: 'rgba(0, 0, 0, 0.06)',
  disabled: 'rgb(201, 201, 201)',
  divider: 'rgb(218, 218, 218)',
  error: '#fe4739',
  facebook: '#3A5497',
  fontSizeLarge: '0.9375rem',
  fontSizeMedium: '0.875rem',
  fontSizeSmall: '0.8125rem',
  github: '#181515',
  glass: 'rgba(0, 0, 0, 0.04)',
  info: '#00c8de',
  instagram: '#F73C8C',
  link: 'rgb(24, 118, 209)',
  neutral: '#46505c',
  odnoklassniki: '#EE8108',
  primary: '#811af0',
  secondary: '#475979',
  skeletonDark: 'rgb(243, 243, 243)',
  skeletonLight: 'rgb(236, 235, 235)',
  sky: '#abcfe5',
  success: '#2cd252',
  telegram: '#23A0DB',
  textOpposite: '#fff',
  textPrimary: '#000000',
  textSecondary: '#6a6a6a',
  twitter: '#1E96E8',
  viber: '#7D5FF6',
  vk: '#0173F6',
  warning: '#f8e100',
  whatsapp: '#19D741',
  white: '#fff',
  youtube: '#FF0200',
};
module.exports = { initialVariables };
